### PR TITLE
remove sites when tab is closed

### PR DESCRIPTION
**Pull request by jdorweiler**
**Reviewer:**

## Description:

We're creating site objects in the backend and they hang around until you restart the browser. There is no reason for these to exist longer than the tab so let's clean them up when the tab is destroyed. 

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [x] **Ensure the PR solves the problem**
- [x] **Review every line of code**
- [x] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

**Comment by jdorweiler** [comment]
I got rid of the Sites container and instead letting the tab create new site objects. 

There were some problems with the https whitelist since we weren't keeping a container of site objects around anymore. To fix this I added another stored whitelist in settings `HTTPSwhiteListed` and updated the `site.setWhitelisted` method to take a whitelist name and value.

The added benefit here is that now the https whitelisted sites will be synced to storage and stay whitelisted after restarting the browser.

File: js/background.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ chrome.webRequest.onBeforeRequest.addListener(
               thisTab.addToPotentialBlocked(tracker.url);
               
               // Block the request if the site is not whitelisted
-              if (!thisTab.site.whiteListed) {
+              if (!thisTab.site.whitelisted) {
```
Comment (russellholt) [inline js/background.js]: 👍

File: test/tests/sites.js [hunk 2]
```diff
@@ -6,7 +6,7 @@
       settings.updateSetting('whitelist', '');
 
       var domain = "test.com";
-      var newSite = Sites.add(domain);
+      var newSite = new Site(domain);
```
Comment (russellholt) [inline test/tests/sites.js]: 👍

File: js/site.js [hunk 3]
```diff
@@ -1,64 +1,74 @@
...
-            this.setWhitelisted(false);
-        }
+        let globalWhiteLists = ['whitelisted', 'HTTPSwhitelisted'];
+
+        globalWhiteLists.map((name) => {
+            let list = settings.getSetting(name) || {};
+            
+            if(list[this.domain]){
+                this.setWhitelisted(name, true);
+            }
```
Comment (russellholt) [inline js/site.js]: isn't this if/else block the same as 

```
this.setWhitelisted(name, list[this.domain]);
```

Comment (jdorweiler) [inline js/site.js]: It is!

File: js/site.js [hunk 4]
```diff
@@ -1,64 +1,68 @@
...
-        let globalWhitelist = settings.getSetting('whitelist') || {};
+        let globalWhiteLists = ['whitelisted', 'HTTPSwhitelisted'];
 
-        if(globalWhitelist[this.domain]){
-            this.setWhitelisted(true);
-        }
-        else{
-            this.setWhitelisted(false);
-        }
+        globalWhiteLists.map((name) => {
```
Comment (russellholt) [inline js/site.js]: can we stick with the change to lowercase in whitelist generally

**Comment by russellholt** [comment]
LGTM

**Pull request merged**